provider/internal/queue/queue.go

```diff
@@ -79,8 +84,6 @@ func (q *Queue) work() {
 		defer func() {
 			// also cancels any in-progess enqueue tasks.
 			q.close()
-			// unblocks anyone waiting
-			close(q.dequeue)
```

Comment:
Added a comment:
```go
	// We don't close dequeue because the provider which consume this get caught in
	// an infinite loop dequeing cid.Undef if we do that.
	// The provider has it's own select on top of dequeue and will handle this by itself.
```